### PR TITLE
Membership Payment updates

### PR DESCRIPTION
**Pull request by bcongdon**
- Added success page for after a member has paid
- Changed payment logic so a user doesn't need to be logged in to pay for membership

Marking as WIP because I still need to do an end-to-end test with the AD service

### CODE REVIEW

**Review by sameetandpotatoes — APPROVED** [review]
lgtm

File: app/controllers/credits.js [hunk 1]
```diff
@@ -207,25 +237,27 @@ module.exports = function(app){
...
+        description: "ACM Membership Purchase",
         adjust_balance: false // Don't give the user credit for this transaction
       }
     }, function(err, response, body){
       if(err) {
         return res.status(500).send(err);
       }
       if(response.statusCode != 200 || !body.successful){
-        req.flash('error', "Something went wrong with your payment. Talk to someone in ACM Admin.");
+        req.flash('error', "Something went wrong with your payment. Please talk to someone in ACM Admin.");
```
Comment (narendasan) [inline app/controllers/credits.js]: Add an email here or link to the #userhelp slack channel

File: app/views/credits/credits_membership_success.ejs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Illinois/NCSA Open Source License.  You should have received a copy of
+this license in a file with the distribution.
+-->
+
+<%- include('../_partials/header') -%>
+
+<div id="meme-upload-container">
+  <div class="row col-centered small-12 medium-6 large-6">
+    <h3>ACM Membership Purchase Confirmation</h3>
+    <p>Thanks for submitting your purchase for an ACM membership! It may take a little bit to get your membership registered on our end, so we appreciate your patience. Once your membership processing is complete, you'll be able to access our intranet <a href="/intranet">here</a>.</p>
```
Comment (narendasan) [inline app/views/credits/credits_membership_success.ejs]: change "you'll be able to access our intranet" to "you'll be able to access ACM's services" or something

**Review by narendasan — CHANGES_REQUESTED** [review]

**Comment by narendasan** [comment]
@aashishkapur did you ever finish the AD service, would be nice to get this in before the end of the year

**Comment by bcongdon** [comment]
Agreed...

**Comment by aashishkapur** [comment]
Yeah, the service has been up for a while now, acm-uiuc/groot-users-service#22 needs to be merged

**Comment by narendasan** [comment]
Looks like thats held up on copy needed @acm-uiuc/top4 is there anything else needed for the merge of the feature?

**Pull request closed without merging**